openupgrade_scripts/scripts/stock/16.0.1.1/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,214 @@
+stock        / stock.production.lot     / ref (char)                    : DEL
+stock        / stock.production.lot     / website_message_ids (one2many): DEL relation: mail.message
+# NOTHING TO DO: handle by ORM
+
+stock        / stock.quant              / lot_id (many2one)             : relation is now 'stock.lot' ('stock.production.lot') [nothing to do]
+# NOTHING TO DO
+
+stock        / stock.quant              / storage_category_id (many2one): NEW relation: stock.storage.category, isrelated: related, stored
+# DONE: pre-migration: create column and fill values
+
+stock        / stock.quant.package      / pack_date (date)              : NEW hasdefault: default
+# NOTHING TO DO
+
+stock        / stock.route              / active (boolean)              : NEW hasdefault: default
+stock        / stock.route              / categ_ids (many2many)         : NEW relation: product.category
+stock        / stock.route              / company_id (many2one)         : NEW relation: res.company, hasdefault: default
+stock        / stock.route              / name (char)                   : NEW required
+stock        / stock.route              / packaging_ids (many2many)     : NEW relation: product.packaging
+stock        / stock.route              / packaging_selectable (boolean): NEW
+stock        / stock.route              / product_categ_selectable (boolean): NEW
+stock        / stock.route              / product_ids (many2many)       : NEW relation: product.template
+stock        / stock.route              / product_selectable (boolean)  : NEW hasdefault: default
+stock        / stock.route              / rule_ids (one2many)           : NEW relation: stock.rule
+stock        / stock.route              / sequence (integer)            : NEW hasdefault: default
+stock        / stock.route              / supplied_wh_id (many2one)     : NEW relation: stock.warehouse
+stock        / stock.route              / supplier_wh_id (many2one)     : NEW relation: stock.warehouse
+stock        / stock.route              / warehouse_ids (many2many)     : NEW relation: stock.warehouse
+stock        / stock.route              / warehouse_selectable (boolean): NEW
+# NOTHING TO DO: handle by ORM
+
+stock        / stock.rule               / location_dest_id (many2one)   : NEW relation: stock.location, required
+stock        / stock.rule               / location_id (many2one)        : DEL relation: stock.location, required
+# DONE: pre-migration: renamed columns
+
+stock        / stock.rule               / route_id (many2one)           : relation is now 'stock.route' ('stock.location.route') [nothing to do]
+stock        / stock.scrap              / lot_id (many2one)             : relation is now 'stock.lot' ('stock.production.lot') [nothing to do]
+# NOTHING TO DO
+
+stock        / stock.scrap              / scrap_qty (float)             : now a function
+# NOTHING TO DO: handle by ORM
```

Comment:
@hoangtiendung070797 ORM sẽ không xử chỗ này
chỗ này cần dựa vào logic compute để update số lượng cho những bản ghi có `move_id`

Reply:
@hoangtiendung070797 cái này anh có xem lại, thấy có `readonly=False`, nên không cần sửa lại nhé, nothing to do nhé

Reply:
@royleviindoo dạ vâng anh, e vừa định push lại lên xong :smile: